crates/driver/src/domain/competition/mod.rs

```diff
@@ -103,19 +103,44 @@ impl Competition {
             }
         });
 
+        // Merge solutions
+        let mut merged: Vec<Solution> = Vec::new();
```

Comment:
IMO keeping this in a separate function is nicer since the function is already complex and the surrounding function quite long as well.